docs/book/stacks-and-components/stack-deployment/cloud-stacks/minimal-gcp-stack.md

```diff
@@ -0,0 +1,179 @@
+
+* Cloud Functions API
+* Cloud Build API
+* Artifact Registry API
+* Cloud Run Admin API
+* Cloud Logging API
+
+### 3) Create a dedicated service account
+
+The service account should have these following roles.
+
+* AI Platform Service Agent
+* Storage Object Admin
+
+These roles give permissions for full crud on storage objects and full permissions for compute within vertex.
+
+### 4) Create a JSON Key for your service account
+
+This [json file](https://cloud.google.com/iam/docs/keys-create-delete) will allow the service account to assume the identity of this service account. You will need the filepath of the downloaded file in the next step.
+
+```bash
+export JSON_KEY_FILE_PATH=<JSON_KEY_FILE_PATH>
+```
+
+### 5) Create a Service Connector within ZenML
+
+The service connector will allow ZenML and other ZenML components to authenticate themselves with GCP.&#x20;
+
+{% tabs %}
+{% tab title="CLI" %}
+```bash
+zenml integration install gcp \
+&& zenml service-connector register gcp_connector \
+--type gcp \
+--auth-method service-account \
+--service_account_json=@<FILE_PATH> \
+--project_id=<GCP_PROJECT_ID>
+```
+{% endtab %}
+
+{% tab title="Dashboard" %}
+<figure><img src="../../../.gitbook/assets/GCP_Service_Connector.png" alt=""><figcaption><p>Choose the GCP Connector</p></figcaption></figure>
+
+<figure><img src="../../../.gitbook/assets/GCP_Connector_Key.png" alt=""><figcaption><p>Paste the entire contents of the key.json here</p></figcaption></figure>
+
+<figure><img src="../../../.gitbook/assets/GCP_Connector_Resources.png" alt=""><figcaption><p>Make sure GCR, GCS and Generic GCP Resources are all selected here.</p></figcaption></figure>
+{% endtab %}
+{% endtabs %}
+
+### 6) Create Stack Components
+
+#### Artifact Store
+
+Before you run anything within the zenml CLI, head on over to GCP and create a GCS bucket, in case you don't have already one that you can use. Once this is done, you can create the zenml stack component as follows:
+
+{% tabs %}
+{% tab title="CLI" %}
+```bash
+export ARTIFACT_STORE_NAME=gcp_artifact_store
+
+# Register the GCS artifact-store and reference the target GCS bucket
+zenml artifact-store register ${ARTIFACT_STORE_NAME} --flavor gcp \
+    --path=gs://<YOUR_BUCKET_NAME>
+
+# Connect the GCS artifact-store to the target bucket via a GCP Service Connector
+zenml artifact-store connect ${ARTIFACT_STORE_NAME} -i
+```
+
+{% hint style="info" %}
+Head on over to our [docs](../../component-guide/artifact-stores/gcp/) to learn more about the registration of an artifact store.
```

Comment:
to learn more about the registration or to learn more about other options and authentication or... somehow 'about the registration' feels off?